baseline-network-linux-hld.md

```diff
@@ -0,0 +1,163 @@
+able to shared common implementations to support open networking hardware.
+
+Support for open networking hardware platforms is currently distributed over
+various different projects. Many platforms, especially older ones are supported
+by OpenNetworkLinux, many are supported by SONiC, and some are supported by
+DentOS, which is based on OpenNetworkLinux, but targets different devices. The
+open hardware support is usually very tightly integrated in the build systems,
+which makes sharing and reusing the support hard.
+
+## Requirements and Dependencies
+
+## Functional Description
+
+### Base Open Networking Hardware Support
+
+The base networking hardware support combines hardware support from multiple
+sources, and therefore will need to support multiple kinds of APIs.
+
+Possible APIs are:
+
+- SONiC PDDF, due to the large amount of available platforms
+  https://github.com/sonic-net/SONiC/blob/master/doc/platform/brcm_pdk_pddf.md
+- S3IP as described in https://github.com/sonic-net/SONiC/pull/1068
+- ONLP, as used by ONL and DentOS
+
+The base networking hardware support is organized in multiple repositories:
+
+- kernel: Linux fork with required base changes (backports of important
+submissions, etc)
+- platform-support: drivers, libraries and userspace code for initializing
+platforms
+(do we need more?)
+- TBD: organization of platform support
+- TBD: platform initialization
+
+In the first iteration, the platform support is fine as-is. Future iterations
+should focus on improving the platform drivers and support code to follow modern
+code style and compile with as few warnings as possible, ideally none.
+
+New platform support should follow modern coding styles from the beginning (e.g.
+kernel coding style for modules).
+
+## Design
+
+![example layering with operating layers on top like meta-dent, the proposed
+hardware support layer in the middle, and base Yocto and Open Embedded layers
+at the bottom](images/layering-example.png "layering example")
+
+### Yocto based example and testing OS
+
+To allow quick testing and easy extension for other users, an example
+implementation and support layer for Yocto will be provided. It will contain all
+basic building blocks for creation of an installable image:
+
+- recipes for above kernel and platform support
+- class and image recipe for a ONIE installable image
+- machine definitions as generic OS-based
+
+### Kernel recipe and configuration handling
+
+The kernel configuration will be done via
+https://docs.yoctoproject.org/kernel-dev/advanced.html to allow large
+flexibility in how other projects using the layer can customize the kernel.
+
+TBD: should we have patches or branches with backport commits? Maybe both?
+
+- patches make required changes due to upstream changes more obvious (provide a
+“conflict diff”) and easier to review
+- branches are closer to the upstream approach, but may be harder to track
+  changes needed due to upstream changes
```

Comment:
Not sure I understand this. Should it be: "but _it_ may be harder"? Otherwise, "may" refers to branches and then the rest of the sentence doesn't work.